src/extension.ts

```diff
@@ -611,6 +615,33 @@ function launchMetals(
         codeLensRefresher
       );
 
+      const getTestUI = () =>
+        getValueFromConfig<TestUIKind>(
+          config,
+          "testUserInterface",
+          "Test Explorer"
+        );
+
+      const istTestManagerDisabled = getTestUI() === "Code Lenses";
+      const testManager = createTestManager(client, istTestManagerDisabled);
+
+      const disableTestExplorer = workspace.onDidChangeConfiguration(() => {
+        const testUI = getTestUI();
+        if (testUI === "Code Lenses") {
+          testManager.disable();
+        } else {
+          testManager.enable();
+        }
+      });
+
+      const refreshTests = client.onRequest(CodeLensRefreshRequest.type, () => {
+        testManager.discoverTestSuites();
+      });
```

Comment:
Test explorer should be refreshed just like code lense. This change is possible due to https://github.com/scalameta/metals/pull/3355